### PR TITLE
Replace kdf argon2id 1527

### PR DESCRIPTION
**Pull request by vitoreiji**

### CODE REVIEW

File: packages/tutanota-argon2id/lib/index.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+let argon2Module = undefined
+
+async function loadArgon2Module() {
```
[the remark this answers is not in the document]
Comment (paw-hub) [reply packages/tutanota-argon2id/lib/index.js]: It's exported now and should be publicly accessible.

[the remark this answers is not in the document]
Comment (charlag) [reply packages/tutanota-argon2id/lib/index.js]: that's not the point. We should let the users decide how to load WASM module. Right now you just have one hardcoded path to it.

[the remark this answers is not in the document]
Comment (paw-hub) [reply packages/tutanota-argon2id/lib/index.js]: I'm really not sure what you mean here, unless you mean passing a function for loading the module instead?

File: buildSrc/DevBuild.js [hunk 2]
```diff
@@ -196,6 +195,12 @@ export async function prepareAssets(stage, host, version) {
 		fs.copy(path.join(root, "/src/braintree.html"), path.join(root, "build/braintree.html")),
 	])
 
+	const wasmDir = path.join(root, "/build/wasm")
+	await Promise.all([
+		await fs.emptyDir(wasmDir),
+		fs.copy(path.join(root, "/packages/tutanota-argon2id/lib/wasm/argon2.wasm"), path.join(wasmDir, "argon2.wasm")),
```
[the remark this answers is not in the document]
Comment (charlag) [reply buildSrc/DevBuild.js:201]: I think this plugin in `file` mode is what we want
https://github.com/mitschabaude/esbuild-plugin-wat

File: packages/tutanota-argon2id/lib/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (charlag) [reply packages/tutanota-argon2id/lib/index.js]: I think we could expose something like `createArgon2Module()`  that accepts a WASM module. We can still provide `loadArgon2Module()` but not force anyone to use it.

File: buildSrc/DevBuild.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (vitoreiji) [reply buildSrc/DevBuild.js:201]: You would have to come up with a solution for the dist build to use that too, then. We'll keep using `fetch`, then you can adapt the build later :)

**Comment by paw-hub** [comment]
Merged via https://github.com/tutao/tutanota/pull/5794

**Pull request closed without merging**